.github/workflows/ci.yml

```diff
@@ -4,9 +4,13 @@ on:
   push:
     branches:
       - main
+      - main_0.1
+      - vnext-prototype
   pull_request:
     branches:
       - main
+      - main_0.1
+      - vnext-prototype
```

Comment:
Does this run build pipeline on all branches for evert PR? 

I am assuming a CI job should be executed based on branch it's forked from. This way if there is a build break it won't block other branches.

Reply:
I don't know I understood your question... CI should run any PR that it wants to merge to.

Reply:
CI executes PR against a branch merged with it's changes. In this case it happens to be `main`. So if I have a feature branch I assume CI is executed with my feature branch merged with main. 

I am trying to understand what's the expected behavior after this change? The current CI run associated with this PR doesn't seems to be using `vnext` or `main_0.1` branch. Is there a way to test `ci.yml` changes before merging to main?